File: solution/1334. Find the City With the Smallest Number of Neighbors at a Threshold Distance/1334.cpp
class Solution {
public:
    int findTheCity(int n, vector<vector<int>>& edges, int distanceThreshold) {
        vector<vector<int>> dist(n, vector<int>(n, INT_MAX));
        for (auto& e : edges) {
            int a = e[0], b = e[1], w = e[2];
            dist[a][b] = min(dist[a][b], w);
            dist[b][a] = min(dist[b][a], w);
        }
        
        for (int i = 0; i < n; ++i)
            dist[i][i] = 0;
        
        for (int k = 0; k < n; ++k) {
            for (int i = 0; i < n; ++i) {
                for (int j = 0; j < n; ++j) {
                    if (dist[i][k] == INT_MAX || dist[k][j] == INT_MAX) continue;
                    if (dist[i][k] + dist[k][j] < dist[i][j])
                        dist[i][j] = dist[i][k] + dist[k][j];
                }
            }
        }
        
        int mn = INT_MAX, res = 0;
        vector<int> count(n);
        for (int i = 0; i < n; ++i) {
            int cnt = 0;
            for (int j = 0; j < n; ++j) {
                if (i == j) continue;
                if (dist[i][j] <= distanceThreshold)
                    ++cnt;
            }
            if (cnt <= mn) {
                mn = cnt;
                res = max(res, i);
            }
        }
        return res;
    }
};

File: solution/1334. Find the City With the Smallest Number of Neighbors at a Threshold Distance/1334.py
class Solution:
    def findTheCity(self, n: int, edges: List[List[int]], distanceThreshold: int) -> int:
        dist = [[float('inf')] * n for _ in range(n)]
        
        for i, j, w in edges:
            dist[i][j] = dist[j][i] = w
            
        for i in range(n):
            dist[i][i] = 0
            
        for k in range(n):
            for i in range(n):
                for j in range(n):
                    dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])
        
        res = {sum(d <= distanceThreshold for d in dist[i]): i for i in range(n)}
        return res[min(res)]
    
